editor/gui/editor_spin_slider.cpp

```diff
@@ -142,8 +130,44 @@ void EditorSpinSlider::gui_input(const Ref<InputEvent> &p_event) {
 	}
 
 	Ref<InputEventKey> k = p_event;
-	if (k.is_valid() && k->is_pressed() && k->is_action("ui_accept", true)) {
-		_focus_entered();
+	if (k.is_valid() && k->is_pressed()) {
+		if (k->is_action("ui_accept", true)) {
+			_focus_entered();
+		} else if (k->is_action("ui_cancel", true) && is_grabbing()) {
+			_grab_end();
+			set_value(pre_grab_value);
+		}
```

Comment:
```suggestion
		} else if (is_grabbing()) {
			if (k->is_action("ui_cancel", true)) {
				_grab_end();
				set_value(pre_grab_value);
			}
			accept_event();
		}
```

Reply:
If you want to make this solid, I think you should accept_event() here, to prevent tab or arrow keys from focusing a different UI element, which then breaks the aborting functionality.